--- cmd/calculate/helpers.go
+++ cmd/calculate/helpers.go
@@ -5,6 +5,7 @@
	"database/sql"
	"encoding/json"
	"github.com/similar-manga/similar/internal"
+	"iter"
	"log"
	"os"
	"path/filepath"
@@ -48,19 +49,20 @@
	internal.CheckErr(err)
 }

-func getDBSimilar() []internal.DbSimilar {
-	rows, err := internal.DB.Query("SELECT UUID, JSON FROM SIMILAR ORDER BY UUID ASC")
-	internal.CheckErr(err)
-	defer rows.Close()
-
-	var similarList []internal.DbSimilar
-	for rows.Next() {
-		similar := internal.DbSimilar{}
-		err = rows.Scan(&similar.Id, &similar.JSON)
-		internal.CheckErr(err)
-		similarList = append(similarList, similar)
+func getDBSimilar() iter.Seq[internal.DbSimilar] {
+	return func(yield func(internal.DbSimilar) bool) {
+		rows, err := internal.DB.Query("SELECT UUID, JSON FROM SIMILAR ORDER BY UUID ASC")
+		internal.CheckErr(err)
+		defer rows.Close()
+
+		for rows.Next() {
+			similar := internal.DbSimilar{}
+			err = rows.Scan(&similar.Id, &similar.JSON)
+			internal.CheckErr(err)
+			if !yield(similar) {
+				return
+			}
+		}
+		internal.CheckErr(rows.Err())
	}
-	internal.CheckErr(rows.Err())
-	return similarList
 }

 func WriteLineToDebugFile(fileName string, line string) {
